### ISSUE TITLE
Talon syntax highlighting

### ISSUE THREAD

**Issue by C-Loftus** [TalonCommunity/Wiki#255]
If someone is good at regex it would greatly help us and make the wiki nicer if we could port vscode format from https://github.com/mrob95/vscode-TalonScript/blob/master/syntaxes/talon.tmLanguage.json into the prism syntax.

All the highlighting just goes within the `prism-talon.js` file and uses the syntax as defined by https://prismjs.com/extending.html

Since Prism uses a different syntax unfortunately most content from the vscode addon has to be edited a little bit. However, the vscode addon is a very good start.

If anyone is good at regex and wants to help, that would be super helpful.  Ping me if you have any questions and want to help.

**Comment by Nova38** [comment]
There might also be the potential to use the vscode theme directly via shikjs https://github.com/shikijs which uses the same render as vscode. It might need a little work to get it to work, but it does have a official rehype plugin I think.

It's uses is also discussed here:
https://github.com/facebook/docusaurus/issues/9122

**Comment by C-Loftus** [comment]
Looks cool, not sure how much bandwidth that will take. We have most of the logic for syntax highlighting so I am thinking it might just be easier to port the existing regexes to the prism syntax.  But definitely good to know and if someone does want to tackle this at some point, certainly fine with whatever is easier.

**Comment by Nova38** [comment]
Fair enough, I will take a shot at it here in a week or two. I'm settling into a new job. I am also am gonna be finishing up an addition to the talondoc repo to use the treesitter bindings she made to get syntax highlighting working with pygments and sphinx.bIts mostly done from a year or so. After that I can probably give this a shot.

**Comment by Nova38** [comment]
I have been playing around with getting the prism.js language definition to work, but I am not sure that it will be anywhere near as good as the textmate Grammer due to the limitations of the prisms.js parser. To make it I ended up modifying the existing python Grammer, it uses JavaScript Regex as apposed to the one used by the textmate which have a lot of drawbacks. Most importantly it doesn't handle the context header part of the document well. The textmate Grammer that we have is also a lot better at handling parseing the Regex expressions.



I am working on trying to get the shiki library to work. It is looking promising. There is a small bug upstream in docusouras with they way they are loading modules with jiti that is being worked on in the issue I mentioned above. Ones that is done it should be mostly working. Someone else got it working here https://lachieh.github.io/docusaurus-with-shiki-rehype/docs/intro.


Also I have gotten the textmate gramer included in the shiki repo where they have the included languages. It was pretty easy as I had already gotten it included in GitHub's linquist a year and a half ago or so. 

They also recently added a feature to colorizez brackets like in vscode so that it is easier to see the pairs of them. I think that would be expesially helpful when it comes to the brackets used in TalonScript for the rules.
https://shiki.style/packages/colorized-brackets

**Comment by Nova38** [comment]
It looks like the docusourus team might be waiting a bit to update to the newer loader that would need to be to fix an error when loading it 

Another option to get this would be to migrate to a documentation system that natively uses shiki as it's highlighter as talonscript is supported by shiki out of the box now.


One option for that that is similar to the current setup is starlight which is the documentation framework built on and used by astro. https://starlight.astro.build/getting-started/I have a fork where I played around with converting it and it didn't seem to hard. I would be willing to finish the conversation in my branch if you would like to see how it would end up looking. The main thing is changing the markdown files to. Have the title in the yaml front matter instead of of in the document and then moving the side at location overrides to the front matter as well.


- It's code components are provided via expressive code which wrapped shiki with a few additional features that could be nice for the documentation 
	- https://expressive-code.com/key-features/code-component/
	- It also added in the ability to annotate the code blocks to highlight parts of it and to also show line additional/changes
https://expressive-code.com/key-features/text-markers/
![Screenshot_20250203_131915_Vivaldi.jpg](https://github.com/user-attachments/assets/24c4c421-c519-4d19-8d0c-1db376ad2688)

![Screenshot_20250203_131845_Vivaldi.jpg](https://github.com/user-attachments/assets/84dd26cb-1b17-4974-a1dc-448d5229121c)


- it also has a few builting components that could be illustrative for different parts of the documentation 
	- For instance it has a file tree component that could be used in the getting started/basic configuration to show the talon user directory
		- https://starlight.astro.build/components/file-tree/
![Screenshot_20250203_131723_Vivaldi.jpg](https://github.com/user-attachments/assets/b1dcf6a5-517b-4d7a-9de8-7a7e2c49250a)
